content/concepts/hot-module-replacement.md

```diff
@@ -6,52 +6,87 @@ contributors:
 - sokra
 ---
 
-Hot Module Replacement (HMR) exchanges, adds, or removes modules while an
-application is running without a page reload. You basically can update changed modules without a full page reload.
+Hot Module Replacement (HMR) exchanges, adds, or removes
+[modules](/concepts/modules/) while an application is running without a
+page reload. This allows you to speed up development time by updating
+individual modules when they are changed without refreshing the page.
 
 ## How Does It Work?
 
 ### From The App View
 
-The app code asks the HMR runtime to check for updates. The HMR runtime downloads the updates (async) and tell the app code that an update is available. The app code asks the HMR runtime to apply updates. The HMR runtime applies the update (sync). The app code may or may not require user interaction in this process (you decide).
+1. The app code asks the HMR runtime to check for updates.
+2. The HMR runtime downloads the updates (asynchronously) and tells the app
+code that an update is available.
+3. The app code then asks the HMR runtime to apply the updates.
+4. The HMR runtime applies the update (synchronously).
+
+You can set up HMR so that this process happens automatically, or you can
+choose to require user interaction for updates to occur.
 
 ### From The Compiler (webpack) View
 
-In addition to the normal assets the compiler need to emit the "Update" to allow updating from previous version to this version. The "Update" contains two parts:
+In addition to the normal assets, the compiler needs to emit an "update"
+to allow updating from previous version to the new version. The "update"
+consists of two parts:
 
-1. the update manifest (JSON)
-2. one or multiple update chunks (JavaScript)
+1. The update manifest (JSON)
+2. One or more update chunks (JavaScript)
 
-The manifest contains the new compilation hash and a list of all update chunks (2.).
+The manifest contains the new compilation hash and a list of all update chunks.
 
-The update chunks contain code for all updated modules in this chunk (or a flag if a module was removed).
+Each update chunk contains code for all updated modules in the respective chunk
+(or a flag if a module was removed).
 
-The compiler addtionally makes sure that module and chunk ids are consistent between these builds. It uses a "records" json file to store them between builds (or it stores them in memory).
+The compiler addtionally makes sure that module IDs and chunk IDs are consistent
+between these builds. It stores these IDs in a "records" JSON file or in memory.
```

Comment:
When using webpack-dev-{middleware,server}, an in-memory filesystem is used instead of the real filesystem. This is *probably* the most common way to use HMR, so perhaps it shouldn't be between parentheses.